lib/wor/requests/base.rb

```diff
@@ -0,0 +1,151 @@
+require 'uri'
+require 'httparty'
+require 'json'
+
+# rubocop:disable ClassLength
+module Wor
+  module Requests
+    class Base
+      VALID_HTTP_VERBS = [:get, :post, :patch, :put, :delete].freeze
+      # According to RFC 7231
+      VALID_COMMON_ATTRIBUTES = [:path, :headers, :attempting_to, :response_type].freeze
+      VALID_GET_ATTRIBUTES    = VALID_COMMON_ATTRIBUTES + [:query]
+      VALID_POST_ATTRIBUTES   = VALID_COMMON_ATTRIBUTES + [:body, :query]
+      VALID_PATCH_ATTRIBUTES  = VALID_COMMON_ATTRIBUTES + [:body, :query]
+      VALID_PUT_ATTRIBUTES    = VALID_COMMON_ATTRIBUTES + [:body, :query]
+      VALID_DELETE_ATTRIBUTES = VALID_COMMON_ATTRIBUTES + [:query]
+
+      def get(opts = {}, &block)
+        request(
+          opts.select { |k, _v| VALID_GET_ATTRIBUTES.include?(k) }.merge(method: :get),
+          &block
+        )
+      end
+
+      def post(opts = {}, &block)
+        request(
+          opts.select { |k, _v| VALID_POST_ATTRIBUTES.include?(k) }.merge(method: :post),
+          &block
+        )
+      end
+
+      def patch(opts = {}, &block)
+        request(
+          opts.select { |k, _v| VALID_PATCH_ATTRIBUTES.include?(k) }.merge(method: :patch),
+          &block
+        )
+      end
+
+      def put(opts = {}, &block)
+        request(
+          opts.select { |k, _v| VALID_PUT_ATTRIBUTES.include?(k) }.merge(method: :put),
+          &block
+        )
+      end
+
+      def delete(opts = {}, &block)
+        request(
+          opts.select { |k, _v| VALID_DELETE_ATTRIBUTES.include?(k) }.merge(method: :delete),
+          &block
+        )
+      end
+
+      # rubocop:disable AbcSize
+      # rubocop:disable MethodLength
```

Comment:
We can do something about this? I mean, at least we can divide the if-else in two more methods, one for failure and other for success.